ledger/block/src/transaction/deployment/mod.rs

```diff
@@ -28,25 +28,39 @@ use console::{
 use synthesizer_program::Program;
 use synthesizer_snark::{Certificate, VerifyingKey};
 
-#[derive(Clone, PartialEq, Eq)]
+#[derive(Clone)]
 pub struct Deployment<N: Network> {
     /// The edition.
     edition: u16,
     /// The program.
     program: Program<N>,
     /// The mapping of function names to their verifying key and certificate.
     verifying_keys: Vec<(Identifier<N>, (VerifyingKey<N>, Certificate<N>))>,
+    /// An optional checksum for the program.
+    /// This field creates a backwards-compatible implicit versioning mechanism for deployments.
+    /// Before the migration height where this feature is enabled, the checksum will **not** be allowed.
+    /// After the migration height where this feature is enabled, the checksum will be required.
+    program_checksum: Option<Field<N>>,
 }
 
+impl<N: Network> PartialEq for Deployment<N> {
+    fn eq(&self, other: &Self) -> bool {
+        self.edition == other.edition && self.verifying_keys == other.verifying_keys && self.program == other.program
+    }
```

Comment:
Note that deployment IDs will be different for the same deployment with and without a checksum.